code/client/launcher/Bootstrap.cpp

```diff
@@ -73,6 +74,19 @@ bool VerifyViability();
 
 extern void ResetUpdateChannel();
 
+void MessageBox_OnFailedRequestURL(int result)
+{
+	switch(result)
+	{
+	case 7: // CURL ERROR 7 Failed to connect, request is blocked by a firewall or similar.

```

Comment:
Can we perhaps amend this to only show the 'custom' message for known-firewall-y _detailed_ error strings? I've mainly seen the 'Bad access' variant show this early, but there might have been others (e.g. TLS stuff).

Reply:
Do you know of a list of these error strings and do we also need to take in account localization with these?

We could also choose to add such instructions (check their firewall or network) as an optional but potential fix. I looked around for error code 7 and it's often associated with a wrong IP, port, or mostly a firewall (or equivalent) on PC/network that's blocking it.